### ISSUE TITLE
Upgrade npm to pnpm

### ISSUE THREAD

**Issue by lakardion** [thinknimble/tn-spa-bootstrapper#239]
As much as a good default npm is, there are better tools out there that leverage caching solutions and make builds significantly faster. 
Yarn was initially a good approach to this and [pnpm](https://pnpm.io/) followed.
Pnpm used to be an immature package manager but it has become very popular amongst library authors becoming a standard in modern js development.

https://pnpm.io/benchmarks

**Comment by paribaker** [comment]
pnpm is not supported by heroku  so we would have to have a different set up for heroku ? @lakardion

**Comment by whusterj** [comment]
@lakardion - This issue is getting stale, but I am interested in any strategy that will reduce build times. Can you revisit this and start a PR?

**Comment by lakardion** [comment]
> [@lakardion](https://github.com/lakardion) - This issue is getting stale, but I am interested in any strategy that will reduce build times. Can you revisit this and start a PR?

Ideally we could try moving to `bun` which appears to be like the "uv" of JS nowadays. 
There are some compatibility issues (with vitest for example) that I had run into back when I tried to do a migration like this with tn-models. But we can see what the state of that is now and check whether we can integrate that in our workflow, it would be a huge boost in our installs/builds

**Comment by whusterj** [comment]
Yeah, I've heard good things about `bun`. That sounds like a plan to me.

**Comment by whusterj** [comment]
stale
